lib/omnibus/s3_license_cache.rb

```diff
@@ -162,16 +162,21 @@ def key_for(software, license_file)
           raise InsufficientSpecification.new(:version, software)
         end
 
-        unless software.hash
-          raise InsufficientSpecification.new(:hash, software)
-        end
-
-        # We add Software#shasum in the cache key. It's an accurate way
+        # We add a custom hash of the software recipe in the cache key. It's an accurate way
         # to know if a software definition changed, as it takes into account the
         # resolved version (ie. the git commit hash if the source is a git repository,
         # the hashsum of the downloaded file if the source is a remote file), the project,
         # and all build commands run in the software definition.
```

Comment:
I think we can fully remove this paragraph since it's pretty much describing what `shasum` does

```suggestion
```

Reply:
Indeed, I rephrased a bit to keep the bits that I think can stay, but I'm also fine with removing it altogether